resolwe_bio/processes/reads_processing/alignmentsieve.py

```diff
@@ -0,0 +1,151 @@
+    category = "Read Processing"
+    data_name = 'Sieved BAM ({{ alignment|sample_name|default("?") }})'
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.CACHED
+    entity = {
+        "type": "sample",
+    }
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {"docker": {"image": "resolwebio/common:1.6.0"}},
+        "resources": {
+            "cores": 10,
+            "memory": 16384,
+            "network": True,
+        },
+    }
+
+    class Input:
+        """Input fields to process AlignmentSieve."""
+
+        alignment = DataField("alignment:bam", label="Alignment BAM file")
+        min_fragment_length = IntegerField(
+            label="--minFragmentLength",
+            description="The minimum fragment length needed for "
+            "read/pair inclusion. This option is primarily useful in "
+            "ATACseq experiments, for filtering mono- or di-nucleosome "
+            "fragments. (Default: 0)",
+            default=0,
+        )
+        max_fragment_length = IntegerField(
+            label="--maxFragmentLength",
+            description="The maximum fragment length needed for "
+            "read/pair inclusion. A value of 0 indicates "
+            "no limit. (Default: 0)",
+            default=0,
+        )
+
+    class Output:
+        """Output fields to process AlignmentSieve"""
+
+        bam = FileField(label="Clipped BAM file")
+        bai = FileField(label="Index of clipped BAM file")
+        stats = FileField(label="Alignment statistics")
+        bigwig = FileField(label="BigWig file", required=False)
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+
+    def run(self, inputs, outputs):
+        """Run code of AlignmentSieve process"""
+
+        basename = Path(inputs.alignment.bam.path).name
+        assert basename.endswith(".bam")
+        name = basename[:-4]
+
+        bam_species = inputs.alignment.species
+        output_bam_file = f"{name}_filtered.bam"
+        filter_metrics = f"{name}_metrics.txt"
+        bam_bigwig = f"{name}_filtered.bw"
+
+        params = [
+            "--verbose",
+            "--bam",
+            inputs.alignment.bam.path,
+            "--outFile",
+            output_bam_file,
+            "--filterMetrics",
+            filter_metrics,
+            "--maxFragmentLength",
+            inputs.max_fragment_length,
+            "--minFragmentLength",
+            inputs.min_fragment_length,
+        ]
+
+        return_code, _, _ = Cmd["alignmentSieve"][params] & TEE(retcode=None)
+
+        if return_code:
+            self.error("Error sieving the BAM file.")
+
+        if not os.path.exists(output_bam_file):
+            self.error(f"File {output_bam_file} not created.")
+
+        outputs.bam = output_bam_file
+        self.progress(0.3)
+
+        stderr_file = f"{name}_stderr.txt"
+        (Cmd["samtools"]["index"][f"{output_bam_file}"] > stderr_file)()
```

Comment:
Standard output and error could be caught using `return_code, stdout, stderr = Cmd["samtools"]["index"][f"{output_bam_file}"] & TEE(retcode=None)` to avoid reading and writing files.

Reply:
Much improved, thanks.